Project2015To2017/Reading/ConditionEvaluator.cs

```diff
@@ -119,31 +119,60 @@ public static void UpdateConditionedPropertiesTable
 
 			// it makes little sense for condition to be that short
 			if (condition.Length < 2)
+			{
 				return res;
+			}
 
 			foreach (var keyValuePair in GetConditionValues(condition))
 			{
 				if (keyValuePair.Value.Count != 1)
+				{
 					continue;
+				}
+
 				res.Add(keyValuePair.Key, keyValuePair.Value[0]);
 			}
 
 			return res;
 		}
 
+		/// <summary>
+		/// Process-lifetime-long cache of evaluated conditions
+		/// <remarks>
+		/// It was supposed to be of <see cref="System.Runtime.Caching.ObjectCache"/> type, but that can still remove items from cache under memory pressure,
+		/// while we still need availability guarantees.
+		/// </remarks>
+		/// </summary>
+		private static readonly Dictionary<int, ConditionEvaluationStateImpl> Cache = new Dictionary<int, ConditionEvaluationStateImpl>();
+
 		public static Dictionary<string, List<string>> GetConditionValues(string condition)
 		{
-			var state = new ConditionEvaluationStateImpl();
+			if (condition.Length > 0)
+			{
+				condition = condition.Trim();
+			}
+
+			var hash = condition.GetHashCode();
```

Comment:
same as in the other one, I would keep this the original type that is the key, having a hashcode in there requires intimate knowlegde of how to do these comparisons while it won't add a lot of benefit